### PR TITLE
[module] Add world_first_announcements module

### PR DESCRIPTION
**Pull request by zach2good**
**_I affirm:_**
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)
- [not working right yet] I've _**tested my code and the things my code has changed**_ since the last commit in the PR, and will test after any later commits

## What does this pull request do?

image

This is a good example for how to use:
- GetVolatileServerVariable
- SetVolatileServerVariable
- player:PrintToArea for global messaging

## Steps to test these changes

Do some of the things listed in the file

### CODE REVIEW

**Comment by zach2good** [comment]
Tested the logic of `checkWorldFirstServerVar`, works nicely

image

image

Just trying to work out why the module overriding isn't working for me locally

**Review by Xaver-DaRed — APPROVED** [review]

**Comment by TeoTwawki** [comment]
oh neat. I think I wanna do this with NMs. 1st to kill.

**Comment by zach2good** [comment]
The homepoint override won't work for me locally, but it works for Xaver. Curious.

Just need to test to 75-level up then it should be good to go

**Comment by N3ckB3ard** [comment]
The quest "Beyond the Sun" would work incredibly well with this! Awesome work! Excited to see what we can create with it

**Comment by kaincenteno** [comment]
is this announcement like the moogle tablet one?

**Comment by zach2good** [comment]
image
image

All tested and working

**Comment by zach2good** [comment]
> is this announcement like the moogle tablet one?

No, this is an announcement to all zones, moogle tablet messages are sent to specific zones/processes to execute stuff

File: modules/custom/lua/world_first_announcements.lua [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+end)
+
+m:addOverride("xi.player.onPlayerLevelUp", function(player)
+    super(player)
+
+    checkWorldFirstServerVar(player,
+        "PLAYER_LEVEL_UP",
+        string.format("%s has been the first player to level up!", player:getName()))
+
+    local levelMilestones = { 10, 20, 30, 40, 50, 60, 70, 75 }
```
Comment (WinterSolstice8) [inline modules/custom/lua/world_first_announcements.lua]: 80+?

**Comment by WinterSolstice8** [comment]
I'm guessing things like first Job Master, first to cap merits, etc dont have hooks yet

Other things like first to get abjurations for era servers, or big ticket items like Ridill, kclub, RMEA, things to put on a TODO list at least

File: modules/custom/lua/world_first_announcements.lua (hunk 1 above)
Comment (zach2good) [reply modules/custom/lua/world_first_announcements.lua]: True dat!

**Review by WinterSolstice8 — APPROVED** [review]

**Comment by zach2good** [comment]
Yeah, those others can be left as an exercise to the reader. Abjurations will be easy, just hook the abj guys onEventFinish.

The others would need new hooks I think

**Pull request merged**